src/redux/actions/index.js

```diff
@@ -145,3 +147,15 @@ export const syncStorageToRedux = () => {
     return Promise.resolve();
   };
 };
+
+// Handle languages switcher
+export const switchLanguages = nativeLanguage => {
+  return async dispatch => {
+    if (nativeLanguage === LANGUAGE_NAMES.EN) {
+      dispatch(setLanguageName(LANGUAGE_NAMES.MG));
+    } else {
+      dispatch(setLanguageName(LANGUAGE_NAMES.EN));
+    }
+    return Promise.resolve;
+  };
```

Comment:
We dont have to use thunk here and it can be simplified further by moving some logic into reducer.
make this action like :
```
export const switchLanguages = () => {
  return {
    type: SWITCH_LANGUAGES,
  };
  
  };
  
  ```

Reply:
then in reducer for nativeLanguage
```
case SWITCH_LANGUAGES
return state=== LANGUAGE_NAMES.EN? LANGUAGE_NAMES.MG:LANGUAGE_NAMES.EN 
```